Improve stratosphere documentation

alistairsellar opened the pull request:
<!--
    Thank you for contributing to our project!

    Please do not delete this text completely, but read the text below and keep
    items that seem relevant. If in doubt, just keep everything and add your
    own text at the top, a reviewer will update the checklist for you.

    While the checklist is intended to be filled in by the technical and scientific
    reviewers, it is the responsibility of the author of the pull request to make
    sure all items on it are properly implemented.
-->

## Description

<!--
    Please describe your changes here, especially focusing on why this pull request makes
    ESMValTool better and what problem it solves.

    Before you start, please read our contribution guidelines: https://docs.esmvaltool.org/en/latest/community/

    Please fill in the GitHub issue that is closed by this pull request, e.g. Closes #1903
-->
This PR adds the required section "Available recipes and diagnostics" to the Autoassess stratosphere recipe.

- Closes part of #1775
- Link to documentation: https://esmvaltool--3114.org.readthedocs.build/en/3114/recipes/recipe_autoassess_stratosphere.html

* * *

## Before you get started

<!--
    Please discuss your idea with the development team before getting started,
    to avoid disappointment or unnecessary work later. The way to do this is
    to open a new issue on GitHub.
-->

- [x] [☝ Create an issue](https://docs.esmvaltool.org/en/latest/community/code_documentation.html#contributing-code-and-documentation) to discuss what you are going to do

## Checklist

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

<!-- The next two lines turn the 🛠 and 🧪 below into hyperlinks -->
[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🛠][1] This pull request has a [descriptive title](https://docs.esmvaltool.org/en/latest/community/code_documentation.html#pull-request-title)
- [x] [🛠][1] [Documentation](https://docs.esmvaltool.org/en/latest/community/code_documentation.html#documentation) is available
- [x] [🛠][1] The [list of authors](https://docs.esmvaltool.org/en/latest/community/code_documentation.html#list-of-authors) is up to date
- [x] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/en/latest/community/code_documentation.html#pull-request-checks) were successful

### [New or updated recipe/diagnostic](https://docs.esmvaltool.org/en/latest/community/diagnostic.html)

- [x] [🧪][2] [Recipe is well documented](https://docs.esmvaltool.org/en/latest/community/diagnostic.html#recipe-and-diagnostic-documentation)

***

To help with the number of pull requests:

-  🙏 We kindly ask you to [review](https://docs.esmvaltool.org/en/latest/community/review.html#review-of-pull-requests) two other [open pull requests](https://github.com/ESMValGroup/ESMValTool/pulls) in this repository

<!--
If you need help with any of the items on the checklists above, please do not hesitate to ask by commenting in the issue or pull request.
-->

remi-kazeroni commented on doc/sphinx/source/recipes/recipe_autoassess_stratosphere.rst:
```diff
@@ -165,7 +181,8 @@ References
 ----------------------
 
-ERA-Interim data (Dee et al., 2011) data can be obtained online from ECMWF and NASA respectively.  Monthly mean zonal mean U and T data are required. CMORized that exists on CEDA-Jasmin or DKRZ (contact Valeriu Predoi (`<valeriu.predoi@ncas.ac.uk>`_) for Jasmin or Mattia Righi (`<mattia.righi@dlr.de>`_ )for DKRZ).
+ERA-Interim data (Dee et al., 2011)
```
If possible, I would recommend following what is done for other recipes. I hope I got the ERA-Interim variable used right.
```suggestion
ERA-Interim (hus - cmorizers/data/formatters/datasets/era_interim.py)
```

remi-kazeroni commented on doc/sphinx/source/recipes/recipe_autoassess_stratosphere.rst:
```diff
@@ -165,7 +181,8 @@ References
 Observations Data sets
```
Could we use the same title as in other pages?
```suggestion
Observations and reformat scripts
```

remi-kazeroni reviewed (commented):
Thanks for tackling that @alistairsellar! I see that this documentation page somehow follows a different template with different titles for sections. For example [model-data](https://esmvaltool--3114.org.readthedocs.build/en/3114/recipes/recipe_autoassess_stratosphere.html#model-data) instead of [variables](https://esmvaltool--3114.org.readthedocs.build/en/3114/recipes/recipe_autoassess_stratosphere.html#model-data) (but that table looks nicer than the usual list of variables in other pages!) or [input and usage](https://esmvaltool--3114.org.readthedocs.build/en/3114/recipes/recipe_autoassess_stratosphere.html#inputs-and-usage) instead of [user settings in recipe](https://docs.esmvaltool.org/en/latest/recipes/recipe_autoassess_landsurface_surfrad.html#user-settings-in-recipe) (also nicer in this page, I think). Question: is it planned or possible to rename these sections following what is done for other recipes? But not necessarily reformat the entire sections which look nice here as they are. I leave the rest of the review to @valeriupredoi, don't really have more comments myself.

alistairsellar commented:
Thanks @remi-kazeroni! Good spot!. I've renamed the section titles, and also re-organise sections to follow what seems to be a more standard order.

Also added ta and ua to the list of ERA-Interim vars.

@valeriupredoi, Remi is leaving the final review to you...

remi-kazeroni commented on doc/sphinx/source/recipes/recipe_autoassess_stratosphere.rst:
```diff
@@ -156,19 +124,37 @@ over to the diagnostic/metric is listed below.
+current implementation ERA-Interim.
+
+
+Observations and reformat scripts
+----------------------
```
Sorry, my last suggestion broke the build...
```suggestion
Observations and reformat scripts
-----------------------------------
```

valeriupredoi commented:
many thanks @alistairsellar - couple wiggles in the docs build:
```
/home/docs/checkouts/readthedocs.org/user_builds/esmvaltool/checkouts/3114/doc/sphinx/source/recipes/recipe_autoassess_stratosphere.rst:144: WARNING: Title underline too short.

Observations and reformat scripts
----------------------
/home/docs/checkouts/readthedocs.org/user_builds/esmvaltool/checkouts/3114/doc/sphinx/source/recipes/recipe_autoassess_stratosphere.rst:144: WARNING: Title underline too short.

Observations and reformat scripts
----------------------
```

Also - we may have to pin sphinx to latest as I have just done in Core - let's see if the tests pass with those two wiggles fixed first :beer:

remi-kazeroni reviewed (approved):
Thanks a lot @alistairsellar! Looks fine to me like this. The documentation page follows the same order as in other pages. All nice and clean 👍 @valeriupredoi you can take a final look and merge whenever you get time for this 🍺

valeriupredoi commented:
on it now, gents..

valeriupredoi reviewed (approved):
spick and span, cheers @alistairsellar and Remi for reviewing :beer:

alistairsellar commented:
Brilliant - thanks guys!
